rfcs/0039-minaform-tool-design.md

```diff
@@ -0,0 +1,120 @@
+
+**Unlike in the original `minaform` design, none of these scripts are used to actually deploy or manage cluster resources. Terraform will be the only tool for starting or stopping networks, coda-network and these helper scripts just help get to that stage faster.**
+
+### New coda-network features
+
+- `keyset deploy` command for uploading keys as secrets to kubernetes to deprecate scripts/testnet-keys.py.
+    - If we go this direction permanently, it could be integrated as a local-exec in terraform. See the below keypair download initContainer for another approach.
+- Keyset, keypair, and genesis download to allow easy retrieval of the keys required for a given genesis ledger, or to retrieve any specific key from google cloud storage.
+    - Tracking issue: https://github.com/MinaProtocol/mina/issues/6550
+    - The mina-automation docker image w/ `coda-network keypair download` would allow for an initContainer in kubernetes to grab a key for a specific container, removing the need for `keyset deploy` entirely.
+- Using ocaml generate-keypair binary or docker image to generate keys instead of relying on js library.
+    - Tracking issue: https://github.com/MinaProtocol/mina/issues/6548
+- More complete genesis-create command with support for vesting schedules and a yaml-based config file for defining genesis ledger requirements.
+    - Tracking Issue: https://github.com/MinaProtocol/mina/issues/6551
+    - Doing this properly is worthwhile, but a combination of bash + jq + the existing `coda-network genesis` command have unblocked deployments and will be used until a complete design + implementation is ready.
+
+### ./scripts/create-network.sh
+
+The first entrypoint to deploying a new network is a new script, `./scripts/create-network.sh <testnet_name>`, which will create a new testnet `.tfvars` file based on the new and improved terraform testnet examples, `qa` and `prod`, and then generate or download a set of keys + ledger to use with this new network. This system of standardizing qa and prod networks, and building all future networks from this shared starting point addresses Pain Point #1.
+
+With the example terraform, `create-network.sh` can now use the `coda-network` tool (aka carey's key service, or the "Testnet SDK") to generate all of the neccessary keypairs for you, as well as the genesis ledger itself. In the initial implementation, this will be handled by the already ready to use but fairly hacky scripts `generate-keys-and-ledger.sh`, but as the coda-network tool becomes more fully featured then the sed+jq steps will be removed and this script will interract directly with coda-network.
+
+The flag `--bake-image=true` will cause the tool to also run `./scripts/bake-image.sh` to build a new docker image with all of the required network configuration and s3-cached files baked-in.
```

Comment:
We discussed for sure but mind enumerating this configuration + s3-cached files within the RFC? To help understand the scope better of what's going in the baked image.